gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Sets;
+import com.google.gcloud.Service;
+import com.google.gcloud.spi.DnsRpc;
+
+import java.io.Serializable;
+import java.util.Set;
+
+/**
+ * An interface for the Google Cloud DNS service.
+ *
+ * @see <a href="https://cloud.google.com/dns/docs">Google Cloud DNS</a>
+ */
+public interface Dns extends Service<DnsOptions> {
+
+  /**
+   * The fields of a project.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@code Dns#getProjectInfo(ProjectGetOption...)}. Project ID is always returned, even if
+   * not specified.
+   */
+  enum ProjectField {
+    PROJECT_ID("id"),
+    PROJECT_NUMBER("number"),
+    QUOTA("quota");
+
+    private final String selector;
+
+    ProjectField(String selector) {
+      this.selector = selector;
+    }
+
+    String selector() {
+      return selector;
+    }
+
+    static String selector(ProjectField... fields) {
```

Comment:
Fine as is, but something to consider.
If you had a package scoped interface:
```java
inteface FieldSelector {
  String selector();
}
```
than you can have one static `selector` method that looks like that:
```java
static String selector(FieldSelector required, FieldSelector... fields) {
  Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
  fieldStrings.add(required.selector());
  for (FieldSelector field : fields) {
   fieldStrings.add(field.selector());
 }
 return Joiner.on(',').join(fieldStrings);
}
```
instead of the multiple copy/paste ones...